### ISSUE TITLE
Apache Phoenix persistence

### ISSUE THREAD

**Issue by pablocastilla** [Particular/NServiceBus#4301]
Hi!

In my projects I would like to move all storage to a "apache environment" like cloudera or horton, but also keep using NSB. How do yo see the option of creating a persistence for phoenix? Join it with a kafka transport and you have NSB in the hadoop world.

Regards

**Comment by timbussmann** [comment]
Hey @pablocastilla 

It's very unlikely that we would be able to provide an official persistence for the mentioned platform within the near future. Is your intention to run the persistence on cloud services or are you thinking about on-premise?

But there is always the option of providing your own subscription storage and we also have a documention on how to write one here: https://docs.particular.net/nservicebus/persistence/authoring-custom
We are happy to help you out and answering your questions in case you decide to write your custom persistence.

What, from your perspective, are the main benefits provided by the "apache environment" over the currently provided persistence options?

**Comment by pablocastilla** [comment]
Hi @timbussmann!

My current situation is the following:
We are in the IoT world and we are moving from relational databases to a big data persistence scenario in which we store and analyze trillions of readings/events. Nowadays the first idea is to keep a relational database (SQL Server or Oracle under NHibernate) for all the structural data and have a Horton or Cloudera cluster for the readings/events. NSB comes into place here to execute all the transactional use cases (for instance: install device, send command to device, register client ... etc). We are very happy with it and with your support.

What I would like is to have the same infrastructure for everything, for that I would need a kafka transport and a phoenix (HBase) persistence for sagas. That way I could get rid of the Oracle or SQL Server, lowering my license cost and the deployment complexity.

I don't know if I am alone in this world with this scenario, but I have seen a lot of +1 for the kafka transport in my request here, maybe there are more people like me.
https://github.com/Particular/NServiceBus/issues/3691 
Also I think these technologies are more popular than RavenDB or Azure storage.

I feel that if you give support for a big data scenario you can be a killing technology in IoT projects.

PD: I have to be onpremise and cloud compatible but independent.

**Comment by timbussmann** [comment]
Thanks for the follow-up @pablocastilla 

As far as I remember from the kafka transport discussion, one of the main pain-points which lead us to the decision to not consider this in the v6 timeframe, was the poor experience under windows. To me it sounds like an adaption of .net core would be quite a game changer in that regards, would you agree with that?

> Also I think these technologies are more popular than RavenDB or Azure storage.

If you consider general usage numbers regardless of language, that's indeed imaginable. Would be interesting to have numbers from .NET developers though?

**Comment by pablocastilla** [comment]
I agree. I feel MS is changing a lot and now they have left the "I will do my thing" policy to "I will take what the community offer". So in the near future you will have more mixed environments, not only .net+sql server. I hope we will see .net running on linux with kafka, hbase and so on. Today is possible. 

Regarding .NET numbers, if you see nuget there are clients, even official from MS, for kafka,hbase, phoenix.... Of course these packages have less downloads than older techonologies, but they are gaining momentum.

Maybe is too soon for this, but I just want to put my needs and my +1 :) .

**Comment by timbussmann** [comment]
From my perspective, this seems to be a very interesting story once we have .net core support covered. It looks like there is much more value in supporting kafka & co. once we are able to provide a proper Linux experience. I'm adding this to the feature backlog, thanks for raising this! 👍
